File: semsar-dotnet/Semsar/Models/chat/Message.cs
namespace Semsar.Models.chat
{
    public class Message
    {
        public required string ChatRoom { get; set; }

        public required string SenderId { get; set; }

        public required string SenderName { get; set; }

        public required string SentMessage { get; set; }

        public required string ReciverId { get; set; }

        public required DateTime SentTime { get; set; }

    }
}

File: semsar-flutter/lib/Models/message.dart
class Message {
  final String chatRoom;

  final String senderId;

  final String reciverId;

  final String message;

  final DateTime sentTime;

  Message({
    required this.chatRoom,
    required this.senderId,
    required this.reciverId,
    required this.message,
    required this.sentTime,
  });

  Message.fromMap(Map<String, dynamic> map)
      : chatRoom = map["chatRoom"] as String,
        senderId = map["senderId"] as String,
        reciverId = map["reciverId"] as String,
        message = map["sentMessage"] as String,
        sentTime = DateTime.parse(map["sentTime"]);
}
